watertap/unit_models/zero_order/coag_and_floc_zo.py

```diff
@@ -208,7 +208,8 @@ def rule_chem_flow(blk, t, j):
             elif j == "polymer":
                 chemical_dosage = blk.polymer_dose[t]
             return blk.chemical_flow_mass[t, j] == pyunits.convert(
-                chemical_dosage * blk.properties[t].flow_vol,
+                chemical_dosage
+                * blk.properties[t].flow_vol,  # don't fully understand this test
```

Comment:
I would appreciate someone else looking at this test. @bknueven do you recall this model?

Reply:
This is fine, if clunky.

That said, I think in many, many places, instead of suppressing `pylint` we should instead add an additional `else` clause which raises an exception. WaterTAP is big enough that we should have our own exceptions anyways.

E.g., in this case the `else` clause could raise a `WaterTAPDeveloperError`, or if we want to be cute, a `FronzenPipes` error. It doesn't even need a message if we know we'll never hit the `else` clause upon inspection.

Reply:
Thank you for taking a look and sounds good-- suppressing was a temporary step and throwing a dev error makes sense.